staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/compilation.go

```diff
@@ -105,3 +104,11 @@ func Compile(s *schema.Structural, isResourceRoot bool) ([]CompilationResult, er
 
 	return compResults, nil
 }
+
+// generateUniqueVariableTypeName creates a randomized placeholder type name to use in a CEL programs for cases
+// where we do not wish to expose a stable type name to CEL validator rule authors. For this to effectively prevent
+// developers from depending on the generated name (i.e. using it in CEL programs), it should be changed each time a
+// CRD is created or updated.
+func generateUniqueVariableTypeName() string {
+	return string(uuid.NewUUID())
```

Comment:
uuid.NewUUID() includes dashes... (e.g. "d5b8381c-8f82-4217-bb53-e59da97c8095")... is that ok for a type name? could also do something like

```go
fmt.Sprintf("apiextensions.k8s.io.v1alpha1.ValidationRuleSelf%d", time.Now().Nanosecond())
```

unique per compiled CEL program certainly accomplishes what we want... could also just init a package-scoped var once per kube-apiserver process lifetime if you wanted... anything to make it an unusable target from within the CEL expressions

Reply:
SG, I've switched to this.